src/main/java/randoop/condition/OutcomeTable.java

```diff
@@ -0,0 +1,109 @@
+package randoop.condition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import randoop.test.ExpectedExceptionGenerator;
+import randoop.test.ExtendGenerator;
+import randoop.test.InvalidCheckGenerator;
+import randoop.test.PostConditionCheckGenerator;
+import randoop.test.TestCheckGenerator;
+
+/** Records the outcome of checking all of the conditions for a method. */
+public class OutcomeTable {
+
+  /** Indicates whether this table is empty. */
+  private boolean isEmpty = true;
+
+  /** Indicates whether this table has a valid entry. */
+  private boolean hasValid = false;
+
+  /** The list of throws clauses for this table. */
+  private final List<Set<ThrowsClause>> exceptionSets;
+
+  /** The list of post-conditions for this table. */
+  private final List<PostCondition> postConditions;
+
+  /** Creates an empty {@link OutcomeTable}. */
+  public OutcomeTable() {
+    exceptionSets = new ArrayList<>();
+    postConditions = new ArrayList<>();
+  }
+
+  /**
+   * Adds the outcome of checking the conditions of a specification.
+   *
+   * @param preconditionsSatisfied boolean value indicating whether all preconditions satisfied
+   * @param throwsClauses set of exception type-comment pairs for exceptions expected in post-state
+   * @param postCondition post-condition that must be true in post-state if no exception is thrown,
+   *     null if none
+   */
+  void add(
+      boolean preconditionsSatisfied,
+      Set<ThrowsClause> throwsClauses,
+      PostCondition postCondition) {
+    isEmpty = false;
+    if (preconditionsSatisfied) {
+      if (postCondition != null) {
+        postConditions.add(postCondition);
+      }
+      hasValid = true;
+    }
+    if (!throwsClauses.isEmpty()) {
+      exceptionSets.add(throwsClauses);
+    }
+  }
+
+  /**
+   * Indicate whether this set of results indicates a definitively invalid pre-state. Occurs when
+   * all preconditions fail and there are no expected exceptions.
+   *
+   * <p>This method will evaluate the current state of the table, but should be called after all
+   * entries are added
+   *
+   * @return true if preconditions of all specifications are unsatisfied, and there are no expected
+   *     exceptions; false, otherwise
+   */
+  public boolean isInvalid() {
+    return !isEmpty && !hasValid && exceptionSets.isEmpty();
+  }
+
+  /**
+   * Constructs the {@link TestCheckGenerator} to apply post call by extending the given generator.
```

Comment:
Please make the specification say what the goal or purpose of the method is, from a client point of view.  Saying it creates an object of a particular class is not helpful; a higher-level description is needed.  Here, the goal is to check whether an invocation satisfies the specification.
The phrase "apply post call" is not clear.